app/pages/patient/patientinfo.js

```diff
@@ -399,6 +423,22 @@ var PatientInfo = React.createClass({
       return 'Diagnosis date needs to be a valid date';
     }
 
+    currentDateObj = currentDateObj || new Date();
+    var birthdayDateObj = sundial.parseFromFormat(birthday, FORM_DATE_FORMAT);
+    var diagnosisDateObj = sundial.parseFromFormat(diagnosisDate, FORM_DATE_FORMAT);
+
+    if (birthdayDateObj > currentDateObj) {
```

Comment:
Was just smoke-testing as I'm preparing to merge things in, and these comparisons aren't *quite* working as intended. I'm allowed to set my birthday or diagnosis date to 2015-06-12 even though it's still 2015-06-11 here in Pacific time. The reason for this is that > is comparing the raw integer values (i.e., ms since 1970-01-01 UTC) of the date objects involved, and even though it's still 2015-06-11 in Pacific time it's already 2015-06-12 in UTC. Without a timezone as a third arg to `sundial. parseFromFormat` (link: [sundial source](https://github.com/tidepool-org/sundial/blob/master/lib/datetimeWrapper.js#L329)), the `birthdayDateObj` and `diagnosisDateObj` are being created as UTC, so `2015-06-12T00:00:00.000Z` is not evaluating as greater than my *UTC* now of `2015-06-12T05:38:10.608Z`.

Since we don't *have* a current timezone to pass into `parseFromFormat` as the third arg (as you know @GordyD, these are difficult to infer!), the best solution *may* be to do a string comparison of ISO date strings, taking something like `birthdayDateObj.toISOString.slice(0,10)` and comparing it to an ISO-formatted date string constructed from the local `currentDateObj`. (Annoyingly, JavaScript date does not allow you to get the local date out easily as an ISO-formatted string, so as far as I can tell this would have to be done by something like `sundial.translateMask(currentDateObj.toDateString(), 'ddd MMM DD YYYY', 'YYYY-MM-DD')`.)

Hmm, thinking aloud here, but another way to do it might be to create the `currentDateObj` in a manner more similar to 
the creation of the Date objects in the tests, e.g.:
```
var now = new Date();
var currentDateObj = new Date(now.getFullYear(), now.getMonth(), now.getDate());
```
Yeah, I think I like option 2, since it guarantees creating a local midnight for the relevant date, although let me know your thoughts @GordyD.